DPG/Tasks/qaMatchEff.cxx

```diff
@@ -420,8 +478,32 @@ struct qaMatchEff {
           //
       }   // end pions only
       //
+      // no primary/sec-d pions
+      if (!((tpPDGCode == 211) && (mcpart.isPhysicalPrimary()))) {
+        // gets the pdg code and finds its index in our vector
+        itr_pdg = std::find(pdgChoice.begin(), pdgChoice.end(), tpPDGCode);
+        if (itr_pdg != pdgChoice.cend())
+          // index from zero, so increase by 1 to put in the right bin (and 0.5 not needed but just not to sit in the edge)
+          pdg_fill = (float)std::distance(pdgChoice.begin(), itr_pdg) + 1.5;
```

Comment:
confirm me if I get properly: here the logic is that you want to keep `0` for unmatched tracks (i.e. fakes) and then above 1 a species per bin, depenging on how many of the `pdgChoice` one you decide to keep at runtime with pdgBins. Am I right?